### DISCUSSION TITLE
Where do I find the encryption key for the config?

### DISCUSSION THREAD

**Discussion by den-mac** [emporia-vue-local/esphome#192] [General]
I'm sorry for asking a noob question but for the life of me I can't figure this one out.

# Enable Home Assistant API
api:
  encryption:
    # Encryption key is generated by the new device wizard.
    key: "<generated_key_from_new_device_wizard>"

I looked at these places, and none have such a thing:
1) the Emporia app on my phone
2) the ESPhome add-on within home assistant
3) the home assistant configs for the existing Emporia integration

I know some or all of those is a stretch, but I'm at a loss. Can someone explain? Thanks!

**Comment by den-mac** [comment]
*maybe* this is what this is talking about? I think I'll need to follow these instructions to generate an API key for this to use....

https://community.home-assistant.io/t/2023-2-esphome-deprecated-api-password-how-to-update-to-encryption-key/528886

**Comment by flaviut** [comment]
When you create a new device in the ESPHome dashboard, the new device wizard will also provide you with a key:

![image](https://github.com/emporia-vue-local/esphome/assets/5213469/fbd75bf9-4877-4b35-8613-63e21d2835a7)

and it will also automatically insert the key into your configuration:

![image](https://github.com/emporia-vue-local/esphome/assets/5213469/8f300592-59c1-43fc-b7ca-dbd7f42a5e1b)
